lib/saml_idp/assertion_builder.rb

```diff
@@ -73,6 +77,81 @@ def fresh
     alias_method :raw, :fresh
     private :fresh
 
+    def encrypt(opts = {})
```

Comment:
Most of the added code in this class seems to be somewhat self-contained. What do you think about moving that into its own encapsulated class?